### ISSUE TITLE
need blueprint verify function for nested mcarrays

### ISSUE THREAD

**Issue by xjrc** [LLNL/conduit#258]
The `blueprint::mcarray::verify` function currently only works for 1D multi-component arrays, and a capability is needed for nested versions of these.  As an example, the first of the following multi-component arrays will work and the second will fail:

```
{
    "x": [ ... ],
    "y": [ ... ]
}
```

```
{
    "a":
    {
        "x": [ ... ],
        "y": [ ... ]
    },
    "b":
    {
        "x": [ ... ],
        "y": [ ... ]
    }
}
```

**Comment by xjrc** [comment]
This will be addressed by pull request #246.

**Comment by cyrush** [comment]
resolved in #246
